### PR TITLE
Refactored RebuildStatus.

### PR DESCRIPTION
**Pull request by zuyu**
Assigned to @hbdeshmukh. Thanks.

This PR wraps the rebuild status into a dedicate struct. It also allows future extension info for the distributed versions.

### CODE REVIEW

File: query_execution/QueryExecutionState.hpp [hunk 1]
```diff
@@ -92,8 +92,15 @@ class QueryExecutionState {
...
-    rebuild_status_[operator_index].second = num_rebuild_workorders;
-    rebuild_status_[operator_index].first = rebuild_initiated;
+    auto search_res = rebuild_status_.find(operator_index);
+    if (search_res != rebuild_status_.end()) {
+      search_res->second.has_initiated = rebuild_initiated;
+      search_res->second.num_pending_workorders = num_rebuild_workorders;
+    } else {
+      RebuildStatus rebuild_status(rebuild_initiated, num_rebuild_workorders);
+
+      rebuild_status_.emplace(operator_index, std::move(rebuild_status));
```
Comment (hbdeshmukh) [inline query_execution/QueryExecutionState.hpp]: I am trying to understand the difference between the ``std::move`` usage here and if you use ``std::foward_as_tuple`` instead. For instance, line 100 can be removed and instead line 102 will look as follows:

``rebuild_status_.emplace(operator_index, std::std::forward_as_tuple(rebuild_initiated, num_rebuild_workorders));``

Do you think any one version is better than another?

**Comment by hbdeshmukh** [comment]
Hi @zuyu 

This looks good to me except a small clarifying question.

File: query_execution/QueryExecutionState.hpp (hunk 1 above)
Comment (zuyu) [inline query_execution/QueryExecutionState.hpp]: If we try to use `forward_as_tuple` (which produces a rvalue expression), it actually looks like as following:

```
      rebuild_status_.emplace(std::piecewise_construct,
                              std::forward_as_tuple(operator_index),
                              std::forward_as_tuple(rebuild_initiated, num_rebuild_workorders));
```

But I personally prefer the existing version, as it is more straightforward.

Comment (hbdeshmukh) [inline query_execution/QueryExecutionState.hpp]: Sounds good.

**Comment by hbdeshmukh** [comment]
LGTM, merging.

**Pull request merged**